packages/core/android/src/main/java/io/embrace/embracewrapper/EmbraceManagerModule.java

```diff
@@ -562,6 +568,110 @@ public void logNetworkClientError(String url,
+                return ErrorCode.UNKNOWN;
+
+            default:
+                return null;
+        }
+    }
+
+    private List<Map<String, Object>> transformListReadableMapToListMap(ReadableArray items) {
+        List<Map<String, Object>> objectMapList = new ArrayList<>();
+
+        try{
+            for (int i = 0; i < items.size(); i++) {
+                ReadableMap readableMap = items.getMap(i);
+
+                if (readableMap != null) {
+                    Map<String, Object> map = readableMap.toHashMap();
+                    // TODO Change when Android/iOS replace time in nano for ms
+                    if (map.containsKey("timestampNanos") && map.get("timestampNanos") instanceof Double) {
+                        double timestampNanos = (Double) map.get("timestampNanos");
+                        map.put("timestampNanos", (long) timestampNanos);
+                    }
+                    objectMapList.add(map);
+                }
+            }
+        }catch(Exception e){
+            Log.e("Embrace", "There was an error in parsing the span event data", e);
+        }
+
+        return objectMapList;
+    }
+
+    @ReactMethod()
+    public void stopSpanWithId(String spanId, String errorCodeString, Promise promise) {
+        try{
+            
+            ErrorCode errorCodeInstance = this.getSpanErrorCodebyString(errorCodeString);
+            promise.resolve(Embrace.getInstance().getReactNativeInternalInterface().stopSpan(spanId, errorCodeInstance));
+        }catch(Exception e){
+            promise.resolve(false);
+        }
+    }
+
+    @ReactMethod()
+    public void addSpanAttributesToSpanId(String spanId, String key, String value, Promise promise) {
+        try{
+            promise.resolve(Embrace.getInstance().getReactNativeInternalInterface().addSpanAttribute(spanId, key, value));
+        }catch(Exception e){
+            promise.resolve(false);
+        }
+    }
+
+    @ReactMethod()
+    public void addSpanEventToSpanId(String spanId, String name, Double time, ReadableMap attributes, Promise promise) {
+        try{
+
+            promise.resolve(Embrace.getInstance().getReactNativeInternalInterface().addSpanEvent(spanId, name, time.longValue(), this.convertToReadableMap(attributes)));
+        }catch(Exception e){
+            promise.resolve(false);
+        }
+    }
+
+    @ReactMethod()
+    public void recordCompletedSpanWithName(String name, Double startTimeNanos, Double endTimeNanos, String errorCodeString, String parentSpanId, ReadableMap attributes, ReadableArray events, Promise promise) {
+        try{
+            ErrorCode errorCodeInstance = this.getSpanErrorCodebyString(errorCodeString);
+
+            promise.resolve(Embrace.getInstance().getReactNativeInternalInterface().recordCompletedSpan(name, startTimeNanos.longValue(), endTimeNanos.longValue(), errorCodeInstance, parentSpanId, this.convertToReadableMap(attributes), this.transformListReadableMapToListMap(events)));
```

Comment:
This method expects millis now starting with 6.4

Reply:
Note: We agreed to use Nanos until iOS change it